minidns-dnssec/src/main/java/de/measite/minidns/dane/DaneVerifier.java

```diff
@@ -0,0 +1,244 @@
+/*
+ * Copyright 2015 the original author or authors
+ *
+ * This software is licensed under the Apache License, Version 2.0,
+ * the GNU Lesser General Public License version 2 or later ("LGPL")
+ * and the WTFPL.
+ * You may choose either license to govern your use of this software only
+ * upon the condition that you accept all of the terms of either
+ * the Apache License 2.0, the LGPL 2.1+ or the WTFPL.
+ */
+package de.measite.minidns.dane;
+
+import de.measite.minidns.AbstractDNSClient;
+import de.measite.minidns.DNSMessage;
+import de.measite.minidns.LRUCache;
+import de.measite.minidns.Record;
+import de.measite.minidns.dnssec.DNSSECClient;
+import de.measite.minidns.dnssec.DNSSECMessage;
+import de.measite.minidns.dnssec.UnverifiedReason;
+import de.measite.minidns.record.TLSA;
+
+import javax.net.ssl.HttpsURLConnection;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLPeerUnverifiedException;
+import javax.net.ssl.SSLSession;
+import javax.net.ssl.SSLSocket;
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.X509TrustManager;
+import javax.security.cert.CertificateEncodingException;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.security.KeyManagementException;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.security.cert.Certificate;
+import java.security.cert.CertificateException;
+import java.security.cert.CertificateFactory;
+import java.security.cert.X509Certificate;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * A helper class to validate the usage of TLSA records.
+ */
+public class DaneVerifier {
```

Comment:
RFC 7673 specifies that this is the correct behavior for SRV, too. (It also specifies a fallback to another RFC that I'll have to read :-S )

Nothing spectacular useful but I was wondering if this is the case - mainly as most users are XMPP clients right now, thus SRV matters. Knowing it does not require special handling was useful IMHO. (I'll have to ask if anyone tried to roll out DANE for XMPP yet. Might be an interesting test)

Am 23. Oktober 2015 12:25:40 MESZ, schrieb Marvin W <notifications@github.com>:
>> +import java.security.MessageDigest;
>> +import java.security.NoSuchAlgorithmException;
>> +import java.security.cert.Certificate;
>> +import java.security.cert.CertificateException;
>> +import java.security.cert.CertificateFactory;
>> +import java.security.cert.X509Certificate;
>> +import java.util.ArrayList;
>> +import java.util.Arrays;
>> +import java.util.List;
>> +import java.util.logging.Level;
>> +import java.util.logging.Logger;
>> +
>> +/**
>> + * A helper class to validate the usage of TLSA records.
>> + */
>> +public class DaneVerifier {
>
>I just skimmed over RFC 7673 (didn't know it yet). I don't think it's
>really related as it mainly describes what has to be done before doing
>DANE verification.
>
>Actually I think the whole RFC is worthless, as it's just describes the
>only possible and obvious usage of DANE after a SRV lookup - and the
>fact that a SRV lookup is involved is completely irrelevant after it
>happened.
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/rtreffer/minidns/pull/23/files#r42851297